commands/channel.go

```diff
@@ -345,32 +350,35 @@ func makeChannelPrivateCmdF(c client.Client, cmd *cobra.Command, args []string)
 }
 
 func renameChannelCmdF(c client.Client, cmd *cobra.Command, args []string) error {
-	var newDisplayName, newChannelName string
+	existingTeamChannel := args[0]
+	newChannelName, _ := cmd.Flags().GetString("name")
+	newDisplayName, _ := cmd.Flags().GetString("display_name")
 
-	if len(args) < 2 {
-		return errors.New("Not enough arguments.")
+	// Both Display and name flags are not entered
+	if newDisplayName == "" && newChannelName == "" {
+		return errors.New("Require atleast one flag to rename team, either 'name' or 'display_name'")
 	}
 
-	channel := getChannelFromChannelArg(c, args[0])
+	channel := getChannelFromChannelArg(c, existingTeamChannel)
 	if channel == nil {
-		return errors.New("Unable to find channel '" + args[0] + "'")
+		return errors.New("Unable to find channel '" + existingTeamChannel + "'")
 	}
 
-	newChannelName = args[1]
-	newDisplayName, errdn := cmd.Flags().GetString("display_name")
-	if errdn != nil {
-		return errdn
+	patchedChannel := &model.ChannelPatch{}
+	if newChannelName != "" {
+		patchedChannel.Name = &newChannelName
 	}
-
-	channelPatch := model.ChannelPatch{Name: &newChannelName}
 	if newDisplayName != "" {
-		channelPatch.DisplayName = &newDisplayName
+		patchedChannel.DisplayName = &newDisplayName
 	}
 
-	if _, response := c.PatchChannel(channel.Id, &channelPatch); response.Error != nil {
-		return response.Error
+	// Using PatchChannel API to rename channel
+	_, response := c.PatchChannel(channel.Id, patchedChannel)
+	if response.Error != nil {
+		return errors.New("Cannot rename team '" + existingTeamChannel + "', error : " + response.Error.Error())
```

Comment:
Same: rename _channel_. Also `existingTeamChannel` contains both team and channel names, so cannot rename channel is not really accurate there. Let's pass the `channel.Name` here instead of existingTeamChannel.